aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -806,7 +798,8 @@ impl AptosVM {
             entry_fn.ty_args(),
         )?;
 
-        if is_friend_or_private && txn_data.required_deposit.is_some() {
+        // The `has_randomness_attribute()` should have been feature-gated in 1.11...
+        if is_friend_or_private && has_randomness_attribute(resolver, session, entry_fn)? {
```

Comment:
why this change? isn't it still the required deposit marking the unbiasable?

Reply:
in 1.11 we are doing 
```
if is_friend_or_private && has_randomness_attribute(resolver, session, entry_fn)? {
```
so 
`if is_friend_or_private && txn_data.required_deposit.is_some() {` was actually wrong and will cause split execution results...

Reply:
e.g., in 1.11, a private entry function with randomness attribute calling randomness will abort when unwraping the block randomness seed.

But with `if is_friend_or_private && txn_data.required_deposit.is_some()`, the execution result becomes ahort with BIASIABLE.

Reply:
hmm, so it's a different fix

Reply:
do i have to pull it out? hopefully not 🙏

Reply:
no, but it probably should be called out. like even if the API is not gonna be cherry picked. this change needs to get in